app/src/main/java/org/mozilla/fenix/settings/wallpaper/Extensions.kt

```diff
@@ -25,3 +27,58 @@ fun List<Wallpaper>.groupByDisplayableCollection(): Map<Wallpaper.Collection, Li
 }.toMap().takeIf {
     it.isNotEmpty()
 } ?: mapOf(Wallpaper.DefaultCollection to listOf(Wallpaper.Default))
+
+/**
+ * The ideal scenario is to have 3 seasonal wallpapers and 2 classic, minding the order.
+ * But in case there are less than 3 or no seasonal wallpapers at all, they are replaced by classic.
+ * Edge cases where there are less then 6 items overall would return less items in the right order.
+ */
+fun List<Wallpaper>.groupForWallpaperOnboarding(): List<Wallpaper> {
```

Comment:
Rather than looping the wallpaper several times, I am wondering what if we first called `groupByDisplayableCollection` which will provide us a mapping of the collection and the list of wallpapers associated to their respective collection. You can then determine if the count of each group matches the desired number of seasonal wallpaper or not, and then take the appropriate number of wallpapers from each group to get the final list of wallpapers to display.

Reply:
That was actually my first approach as well, but I didn't like the implementation. Sadly, I have removed it. 

What I didn't like about it: we can't access the classic wallpapers from the map directly, because the access is done through an object we didn't create. I had to look for it and store it in a separate parameter in a for loop. 

And there might be multiple seasonal groups, so we would have to iterate through them as well. It looked like having collections created more problems than brought benefits. In the context of this particular selector there are just seasonal and classic wallpapers, and we can differentiate them in a list as well.

And lastly, it looks like we can't escape having two loops, because we have two groups of not guaranteed length. One loop might satisfy a single condition (let's say, 3 seasonal) but in case we have just 1 classic for some reason, we would have go back to add another seasonal, that we haven't added yet.

I am open to improving the sorting though. This just looked like the optimal approach given the possibly any input from the backend.

Reply:
I don't really know how the wallpapers are structure. So, I am just gonna think out loud here and see if there is any possible approaches here that will make this code simpler. 
1. Is possible to go through the list of wallpaper and construct 2 lists one for seasonal and another for classic wallpapers?
2. After you have the 2 lists, you can check the count of each list and from there take the appropriate number of wallpapers from each.

Reply:
okay, got it! that actually a much easier approach. updated

Reply:
Thanks, this does make much more easier to understand and maintain.